### PR TITLE
Typed properties

### PR DESCRIPTION
**Pull request by BenMorel**
- added missing typed properties
- added a couple missing constructor type hints

I've focused mostly on properties set in constructors (the most obvious ones) in this PR.
I'll propose more typed properties later on, in particular in entities, but these will need careful review to prevent uninitialized property errors wherever a `null` default value is assumed.

### CODE REVIEW

**Comment by BenMorel** [comment]
Hmm, this triggered a lot of phpstan errors. All legitimate, most notably polymorphic calls that were hidden by untyped properties so far. Not sure how you want to handle these?

File: src/EventListener/RegistrationListener.php [hunk 1]
```diff
@@ -13,10 +13,7 @@
  */
 class RegistrationListener implements EventSubscriberInterface
 {
-    /**
-     * @var TokenGenerator
-     */
-    private $tokenGenerator;
+    private TokenGenerator $tokenGenerator;
```
Comment (stof) [inline src/EventListener/RegistrationListener.php:16]: This should rather use the TokenGeneratorInterface for consistency with other classes

Comment (BenMorel) [reply src/EventListener/RegistrationListener.php:16]: `TokenGenerator` is the type used by the constructor. Should the constructor be changed?

Comment (stof) [reply src/EventListener/RegistrationListener.php:16]: I would say yes. Typehinting the interface is better

Comment (BenMorel) [reply src/EventListener/RegistrationListener.php]: Done!

**Comment by Seldaek** [comment]
Would you mind rebasing this? Then the PHPStan errors should be fixed (I can help with that if needed), but for sure moving to typed props is a goal so it's great to get some more of it done :)

**Comment by BenMorel** [comment]
@Seldaek I just rebased, I'll definitely use some help to fix PHPStan errors!

**Comment by Seldaek** [comment]
Thanks :)

**Comment by BenMorel** [comment]
Thanks for the phpstan fixes! :+1:

**Pull request merged**